### DISCUSSION TITLE
Why not use Illuminate\Support\Traits\Macroable in Illuminate\Database\Eloquent\Builder.php?

### DISCUSSION THREAD

**Discussion by parthiban-aspro** [laravel/framework#54039] [Q&A]
i don't have an idea why Builder.php doesn't use macroable trait?

**Comment by crynobone** [comment]
Because it handles macro slightly differently? 

https://github.com/laravel/framework/blob/8f6a9b03b8e41eedca5f6182d1af2b81ad4be3e6/src/Illuminate/Database/Eloquent/Builder.php#L2015-L2057

**Comment by ghost** [comment]
If the question is: Is Eloquent Builder macroable?, the answer is yes.
